x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -254,7 +254,7 @@ function createGetTransactionById(itemsById: IWaterfallIndex) {
     }
 
     const item = itemsById[id];
-    if (item.docType === 'transaction') {
+    if (item && item.docType === 'transaction') {
```

Comment:
@ogupte The reason this error could happen is because our types are incorrect. We should fix those so accessing `itemsById[id]` always returns an optional.

https://github.com/elastic/kibana/blob/868c7125c64a4b7b25484a307cde4e9aafa7109f/x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts#L24-L26

Should be changed to:

```ts
interface IWaterfallIndex {
  [key: string]: IWaterfallItem | undefined;
}
```

This also affects latest master and the issue could happen again.